### DISCUSSION TITLE
NavigationViewItemSelector line does not appear

### DISCUSSION THREAD

**Discussion by rcohn** [microsoft/WinUI-Gallery#1101] [Q&A]
Hello -

In a WinUI3 desktop app I'm working on, I've been trying to get a separator line to show up in a NavigationView.  If I create the NV statically by explicitly adding NavigationViewItems into NavigationView, while inserting a NavigationViewItemSelector after one of the items, the selector line will show up.

However, if the NavigationViewItems are created dynamically through an ObserverCollection, I have not found a way to get the separator line to show up.  The only sample code I've found in WinUI-Gallery where a NavigationViewItemSeparator is inserted into a NavigationViewItem list are for explicit (non-observer) lists.

I suspect that what I want to do can only be accomplished by incorporating the NavigationViewItemSeparator into a NavigationViewItem DataTemplate.  However, I may be overcomplicating things and just don't know if there is a more straightforward way to get the separator to appear.

Would appreciate any pointers on this.  Please let me know if I need to supply any more info to answer the question.

Thanks very much.
Robert Cohn

**Comment by marcelwgn** [comment]
I think that your proposed solution would be viable, my question would be how you decide when to insert a separator and when not to insert one. If your collection contains "separator" items that have no data but only serve as separation, then a DataTemplateSelector would make this quite easy I think.

**Reply by rcohn** [reply]
Hi @chingucoding  - I did find some references that could help with the implementation.  Here are a couple that look useful: 
[https://stackoverflow.com/questions/61717820/uwp-navigationview-with-two-different-sources](url)
[https://github.com/microsoft/microsoft-ui-xaml/issues/2485 ](url)

As to when to insert the separator, that would be done by logic that is used to build the list.

Appreciate the reply.
Best regards.

**Reply by marcelwgn** [reply]
Great to hear that you found some references that help. Is there anything else I can help you with or is your question answered?

**Reply by rcohn** [reply]
Thank you for your replies.  At the moment, my team decided to not use the separator so I'm deferring trying out the above implementation.  But if they decide otherwise, I know how to proceed.

To answer your question, I think my question has been answered as well as can be expected.

Best regards.
